File: applications/rtkbioscangeometry/rtkbioscangeometry.py
import argparse
import sys
from itk import RTK as rtk

def build_parser():
    parser = argparse.ArgumentParser(
        description="Read Bioscan geometry and write RTK geometry XML file.",
        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
    )
    parser.add_argument(
        "--output", "-o", required=True, help="Output geometry XML file"
    )
    parser.add_argument(
        "--verbose", "-v", action="store_true", help="Enable verbose output"
    )
    rtk.add_rtkinputprojections_group(parser)
    return parser

def process(args_info: argparse.Namespace):
    # Create geometry reader
    bioscanReader = rtk.BioscanGeometryReader.New()
    bioscanReader.SetProjectionsFileNames(rtk.GetProjectionsFileNamesFromArgParse(args_info))
    if args_info.verbose:
        print("Reading Bioscan geometry...")
    bioscanReader.UpdateOutputData()

    # Write geometry
    if args_info.verbose:
        print(f"Writing geometry to {args_info.output}...")
    rtk.write_geometry(bioscanReader.GetGeometry(), args_info.output)
    if args_info.verbose:
        print("Done.")

def main(argv=None):
    parser = build_parser()
    args_info = parser.parse_args(argv)
    process(args_info)

if __name__ == "__main__":
    main()

File: applications/rtkbioscangeometry/rtkbioscangeometry.cxx
/*=========================================================================
 *
 *  Copyright RTK Consortium
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         https://www.apache.org/licenses/LICENSE-2.0.txt
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *=========================================================================*/

#include "rtkbioscangeometry_ggo.h"
#include "rtkGgoFunctions.h"
#include "rtkBioscanGeometryReader.h"
#include "rtkThreeDCircularProjectionGeometryXMLFile.h"

int
main(int argc, char * argv[])
{
  GGO(rtkbioscangeometry, args_info);

  // Create geometry reader
  auto bioscanReader = rtk::BioscanGeometryReader::New();
  bioscanReader->SetProjectionsFileNames(rtk::GetProjectionsFileNamesFromGgo(args_info));
  TRY_AND_EXIT_ON_ITK_EXCEPTION(bioscanReader->UpdateOutputData())

  // Write
  TRY_AND_EXIT_ON_ITK_EXCEPTION(rtk::WriteGeometry(
    const_cast<rtk::ThreeDCircularProjectionGeometry *>(bioscanReader->GetGeometry()), args_info.output_arg))

  return EXIT_SUCCESS;
}
